app-backend/db/src/main/scala/com/azavea/rf/database/OrganizationDao.scala

```diff
@@ -206,4 +215,35 @@ object OrganizationDao extends Dao[Organization] with LazyLogging {
       case None => throw new IllegalArgumentException(s"User not in database: ${subjectId}")
     }
   }
+
+  def addLogo(logoBase64: String, orgID: UUID, dataBucket: String): ConnectionIO[Organization] = {
+    val prefix = "org-logos"
+    val key = s"${orgID.toString()}.png"
+    val logoByte = ApacheBase64.decodeBase64(logoBase64)
```

Comment:
You can just take the right now, because of encoding to `String`, then decoding `Base64` bytes from that string, I think we're taking an extra encode/decode loop.

You don't have to get a `String` out first though:

```diff
diff --git a/app-backend/api/src/main/scala/organization/Routes.scala b/app-backend/api/src/main/scala/organization/Routes.scala
index 4caa21a7..1f73605c 100644
--- a/app-backend/api/src/main/scala/organization/Routes.scala
+++ b/app-backend/api/src/main/scala/organization/Routes.scala
@@ -85,7 +85,7 @@ trait OrganizationRoutes extends Authentication
     authorizeAsync (
       OrganizationDao.userIsAdmin(user, orgID).transact(xa).unsafeToFuture()
     ) {
-      entity(as[String]) { logoBase64 =>
+      entity(as[Array[Byte]]) { logoBase64 =>
         onSuccess(OrganizationDao.addLogo(logoBase64, orgID, dataBucket).transact(xa).unsafeToFuture()) { organization =>
           complete((StatusCodes.Created, organization))
         }
diff --git a/app-backend/db/src/main/scala/com/azavea/rf/database/OrganizationDao.scala b/app-backend/db/src/main/scala/com/azavea/rf/database/OrganizationDao.scala
index aae42e08..6f71de8a 100644
--- a/app-backend/db/src/main/scala/com/azavea/rf/database/OrganizationDao.scala
+++ b/app-backend/db/src/main/scala/com/azavea/rf/database/OrganizationDao.scala
@@ -216,10 +216,10 @@ object OrganizationDao extends Dao[Organization] with LazyLogging {
     }
   }
 
-  def addLogo(logoBase64: String, orgID: UUID, dataBucket: String): ConnectionIO[Organization] = {
+  def addLogo(logoBase64: Array[Byte], orgID: UUID, dataBucket: String): ConnectionIO[Organization] = {
     val prefix = "org-logos"
     val key = s"${orgID.toString()}.png"
-    val logoByte = ApacheBase64.decodeBase64(logoBase64)
+    val logoByte = logoBase64 // ApacheBase64.decodeBase64(logoBase64)
     val logoStream = new ByteArrayInputStream(logoByte)
     val md = new ObjectMetadata()
     val s3 = new AWSAmazonS3Client(new DefaultAWSCredentialsProviderChain)
```

Decoding as `Array[Byte]` and then leaving it as is works just fine. For small files this shouldn't make a difference, just a note for The Future :tm: